### PR TITLE
Rename ARN builders for consistency

### PR DESCRIPTION
**Pull request by viren-nadkarni**
## Motivation

This is a small code quality improvement PR that:
- Renames ARN builders to follow a consistent naming scheme
- and groups them together by service

There are no functional changes.

### CODE REVIEW

File: localstack/utils/aws/arns.py [hunk 1]
```diff
@@ -217,29 +266,49 @@ def stepfunctions_activity_arn(name: str, account_id: str, region_name: str) ->
...
+
+
 def firehose_stream_arn(stream_name: str, account_id: str, region_name: str) -> str:
     pattern = "arn:aws:firehose:%s:%s:deliverystream/%s"
     return _resource_arn(stream_name, pattern, account_id=account_id, region_name=region_name)
 
 
-def es_domain_arn(domain_name: str, account_id: str, region_name: str) -> str:
-    pattern = "arn:aws:es:%s:%s:domain/%s"
-    return _resource_arn(domain_name, pattern, account_id=account_id, region_name=region_name)
```
Comment (viren-nadkarni) [inline localstack/utils/aws/arns.py:242]: Duplicate of `elasticsearch_domain_arn`

File: localstack/utils/aws/arns.py [hunk 2]
```diff
@@ -254,9 +323,9 @@ def kms_alias_arn(alias_name: str, account_id: str, region_name: str):
     return _resource_arn(alias_name, pattern, account_id=account_id, region_name=region_name)
 
 
-def code_signing_arn(code_signing_id: str, account_id: str, region_name: str) -> str:
-    pattern = "arn:aws:lambda:%s:%s:code-signing-config:%s"
-    return _resource_arn(code_signing_id, pattern, account_id=account_id, region_name=region_name)
```
Comment (viren-nadkarni) [inline localstack/utils/aws/arns.py:259]: Dead code

**Review by alexrashed — APPROVED** [review]
🚀 🧹

**Review by dominikschubert — APPROVED** [review]
LGTM

In general I'd prefer more localized utils (e.g. for lambda) but it's a good step in the right direction :+1:

**Pull request merged**